src/metatrain/utils/additive/composition.py

```diff
@@ -69,31 +75,104 @@ def __init__(self, hypers: Dict, dataset_info: DatasetInfo):
         for target_name, target_info in self.dataset_info.targets.items():
             self._add_output(target_name, target_info)
 
+    def _get_dataloader(
```

Comment:
I don't understand why do you need a new dataloader function here?

Reply:
So Filippo says that using `drop_last=True` makes training more stable - so in my mind it's good to create a separate dataloader for the CM